fontforgeexe/pythonui.c

```diff
@@ -232,6 +232,9 @@ static void InsertSubMenus(PyObject *args,GMenuItem2 **mn, int is_cv) {
 
     for ( i=5; i<cnt; ++i ) {
 	unichar_t *submenuu = utf82u_copy(PyUnicode_AsUTF8(PyTuple_GetItem(args, i)));
+	for (unichar_t* p = submenuu; *p != '\0'; p++) {
+	    if (*p == '_') *p = 0x200B; // zero-width space
```

Comment:
care to explain this one a bit more?

Reply:
So that a double underscore isn't shown to the user, we replace it. Otherwise user would see something like «_<ins>H</ins>igher-Order Interpolation».

Reply:
I think I'm missing something here. Why do the compiled-in menu items show fine without having to do this?

Reply:
This looks relevant:

https://github.com/fontforge/fontforge/blob/4af47a671c0c32609e6e4b3395fd26b4639317bd/gdraw/gtextinfo.c#L276

Reply:
Ah-ha @jtanx :)

So the real bug here is:

https://github.com/fontforge/fontforge/blob/4af47a671c0c32609e6e4b3395fd26b4639317bd/fontforgeexe/pythonui.c#L234

[Fixed. Now we're using `utf82u_mncopy` as we always should've been.](https://github.com/fontforge/fontforge/pull/4498/commits/142a37cc6c90a45e6a6a689eebd0c49bd6fc0de2#diff-75edc152c5687a95f25ee0a7ceaa2a2c5e3f92c0d5b8d60c7c01111004658c64L234)

Reply:
So now you're using it, why do you need to do this zero-width space thing and why do you still try to find the mnemonics yourself with `append_mnemonic`?

Reply:
@jtanx Well we have to populate `t_mnemonic` somehow. The `mncopy` version must know which character is the mnemonic. Note how there are two fields at play here, `text` and `mnemonic`. Because everywhere else they're hardcoded, we just set e.g. 
```c
gg[i].text = _("_Go");
gg[i].mnemonic = 'G'; // this line matters!
```

Consider this patch:

```patch
diff --git a/fontforgeexe/charinfo.c b/fontforgeexe/charinfo.c
index 44b6d7544..fd36435e4 100644
--- a/fontforgeexe/charinfo.c
+++ b/fontforgeexe/charinfo.c
@@ -4382,19 +4382,19 @@ return;
        memset(&ubox,0,sizeof(ubox));
        memset(&ulabel,0,sizeof(ulabel));
 
-       ulabel[0].text = (unichar_t *) _("Gl_yph Name:");
+       ulabel[0].text = (unichar_t *) _("Glyp_h Name:");
        ulabel[0].text_is_1byte = true;
        ulabel[0].text_in_resource = true;
        ugcd[0].gd.label = &ulabel[0];
        ugcd[0].gd.pos.x = 5; ugcd[0].gd.pos.y = 5+4; 
        ugcd[0].gd.flags = gg_enabled|gg_visible;
-       ugcd[0].gd.mnemonic = 'N';
+       ugcd[0].gd.mnemonic = 'y';
        ugcd[0].creator = GLabelCreate;
        uhvarray[0] = &ugcd[0];
 
        ugcd[1].gd.pos.x = 85; ugcd[1].gd.pos.y = 5;
        ugcd[1].gd.flags = gg_enabled|gg_visible;
-       ugcd[1].gd.mnemonic = 'N';
+       ugcd[1].gd.mnemonic = 'y';
        ugcd[1].gd.cid = CID_UName;
        ugcd[1].creator = GListFieldCreate;
        ugcd[1].data = (void *) (-2);
```

If you compile and open the «Glyph Info» dialog (<kbd>Ctrl + I</kbd>), you'll see <ins>h</ins> marked as the mnemonic, but note that `y` does indeed work!

So, `utf82u_mncopy` only does a special copy into the `text` field _for the purposes of display of the mnemonic string_. But `mnemonic` still must get set somehow—`utf82u_mncopy` can't do it, it doesn't get a reference to that.

Reply:
I don't buy this. By inspection, `utf82u_mncopy` does not read in `mn` (the second parameter) - it *stores* the mnemonic it stripped out. 

In your example, I would not be surprised if there are flags that control whether or not to use the mnemonic specified or something along those lines.

Even without that part, I still don't understand why you have this zero-width-space logic after having run through `utf82u_mncopy`.

I just ran this branch locally with this patch:

```c
diff --git a/fontforgeexe/pythonui.c b/fontforgeexe/pythonui.c
index 3d36e0089..ed8dd6fe8 100644
--- a/fontforgeexe/pythonui.c
+++ b/fontforgeexe/pythonui.c
@@ -232,11 +232,8 @@ static void InsertSubMenus(PyObject *args,GMenuItem2 **mn, int is_cv, unichar_t*

     for ( i=5; i<cnt; ++i ) {
        // 0xFFFF means mnemonic unset
-       unichar_t t_mnemonic = mnemonic[i-5] == 0xFFFF ? 0 : mnemonic[i-5];
+       unichar_t t_mnemonic = 0;// = mnemonic[i-5] == 0xFFFF ? 0 : mnemonic[i-5];
        unichar_t *submenuu = utf82u_mncopy(PyUnicode_AsUTF8(PyTuple_GetItem(args, i)), &t_mnemonic);
-       for (unichar_t* p = submenuu; *p != '\0'; p++) {
-           if (*p == '_') *p = 0x200B; // zero-width space
-       }

        j = 0;
        if ( *mn != NULL ) {
```

and it seemed to work fine for me - which suggests to me that `append_mnemonic` and this zero-width-space business is not needed.